llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -989,6 +989,14 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
       return Cost * LT.first;
     break;
   }
+  case Intrinsic::vp_smax:
+  case Intrinsic::vp_smin:
+  case Intrinsic::vp_umax:
+  case Intrinsic::vp_umin: {
+    auto LT = getTypeLegalizationCost(RetTy);
+    return LT.first *
+           getRISCVInstructionCost(RISCV::VMAXU_VV, LT.second, CostKind);
```

Comment:
TTI uses different Ops to query the cost of `Intrinsic::umin`, `Intrinsic::umax` etc. in the `getIntrinsicInstrCost()`.
Will we need to use different Ops to query the cost  for different vp-intrinsics here? 
```C++
  case Intrinsic::umin:
  case Intrinsic::umax:
  case Intrinsic::smin:
  case Intrinsic::smax: {
    auto LT = getTypeLegalizationCost(RetTy);
    if (LT.second.isScalarInteger() && ST->hasStdExtZbb())
      return LT.first;
    if (ST->hasVInstructions() && LT.second.isVector()) {
      unsigned Op;
      switch (ICA.getID()) {
      case Intrinsic::umin:
        Op = RISCV::VMINU_VV;
        break;
      case Intrinsic::umax:
        Op = RISCV::VMAXU_VV;
        break;
      case Intrinsic::smin:
        Op = RISCV::VMIN_VV;
        break;
      case Intrinsic::smax:
        Op = RISCV::VMAX_VV;
        break;
      }
      return LT.first * getRISCVInstructionCost(Op, LT.second, CostKind);
    }
    break;
  }
```

Reply:
Given that VMINU_VV, VMAXU_VV, VMIN_VV, and VMAX_VV are currently identical, it seems practical to use one representation for all unless a specific case requires distinguishing them.

Reply:
Should we merge them into a single VMIN_MAX_VV?

Reply:
Do you suggest extending the RISCV opcode list like this?
```diff
+enum RVVOpcode {
+  VMIN_MAX_VV = RISCV::INSTRUCTION_LIST_END+1,
+};
+
```

Reply:
Oh I didn't realize they were the actual opcode names. I just though it was our own enum for cost modeling.